content/documents/configuring-user-application-data-access.mdx

```diff
@@ -0,0 +1,502 @@
+})
+.catch((ex: OrderCloudError) => console.log(ex));`,
+csharp: `using OrderCloud.SDK; \n
+var data = new ApiClient
+{
+  "AppName": "Admin Backoffice",
+  "Active": true,
+  "AccessTokenDuration": 600,
+  "AllowAnySeller": true,
+  "ClientSecret": "supersecureclientsecretstring"
+};
+try {
+  ApiClient response = await ApiClients.Create(data);
+  Console.WriteLine(response)
+} catch(OrderCloudException ex) {
+  Console.WriteLine(ex.Message);
+}
+`
+}}
+/>
+
+## Creating Security Profiles
+Now that we have our applications configured with specific access to our marketplace data using API Clients, we should create several Security Profiles to configure data access for our individual users.
+
+### Create a buyer security profile
+Our buyer security profile will represent the breadth of data access we wish to grant to our buyer users, configured using granular API roles.  Let's create a security profile, name it "Default Buyer" with a self-descriptive ID, and specify the `Shopper` role in the Roles array, which enables all actions of the shopping experience.
+<CodeExample
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/securityprofiles HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8\n
+{
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": [
+    "Shopper"
+  ]
+}`,
+javascript: `import { SecurityProfiles } from "ordercloud-javascript-sdk";\n
+SecurityProfiles.Create({
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": [
+    "Shopper"
+  ]
+})
+.then((securityProfile) => {
+    console.log(securityProfile);
+})
+.catch((ex) => console.log(ex));
+`,
+typescript: `import { SecurityProfiles, SecurityProfile, OrderCloudError } from "ordercloud-javascript-sdk";\n
+SecurityProfiles.Create({
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": [
+    "Shopper"
+  ]
+})
+.then((securityProfile: SecurityProfile) => {
+  console.log(securityProfile);
+})
+.catch((ex: OrderCloudError) => console.log(ex));`,
+csharp: `using OrderCloud.SDK; \n
+var data = new SecurityProfile
+{
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": new ApiRole[] { ApiRole.Shopper }
+};
+try {
+  SecurityProfile response = await SecurityProfiles.Create(data);
+  Console.WriteLine(response)
+} catch(OrderCloudException ex) {
+  Console.WriteLine(ex.Message);
+}
+`
+}}
+/>
+
+### Create a seller security profile
+Next, we can create a security profile for our seller users.  Let's create a security profile, name it "Default Seller" with a self-descriptive ID, and specify the `FullAccess` role in the Roles array, which gives our sellers access to any resource in the API.  As a best practice, you would want to control the roles on a more granular level and only assign roles necessary for the user as a security precaution, but for our example, `FullAccess` will work.  You can find more details on security profiles [here](knowledge-base/security-profiles).
```

Comment:
Can we use the blockquote component or bold text here to emphasize the part about FullAccess not being a best practice?

Reply:
I'm actually just adjusting the article to use more granular roles and explaining why, vs. laying out an example that we don't recommend 😆